src/OrbitQt/TargetLabel.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_TARGET_LABEL_H_
+#define ORBIT_QT_TARGET_LABEL_H_
+
+#include <QLabel>
+
+#include "OrbitClientData/ProcessData.h"
+#include "OrbitGgp/Instance.h"
+
+namespace orbit_qt {
+
+class TargetLabel : public QLabel {
```

Comment:
I think you shouldn't inherit from QLabel here. To me it looks like the label should be a private member instead.

You're trying to enforce invariants by setting the label's properties in the setters of `TargetLabel`. But if you inherit from QLabel, the user of this class will be able to mess with this.

I see two solutions:
- Either make it a private member, so you have full control over the invariance and the user can rely on that.
- Or make it obvious that it's the user's responsibility, by making the setters free functions which take a label as an argument. `SetFileTarget(QLabel* target_label, const std::filesystem::path& path);`

Obviously the first is closer to the current solution.

Reply:
I second this, because it'll also make it easier to extend it by e.g. adding a "Connected" icon as we discussed.

Reply:
I implemented the first solution (in a way). I mark this as resolved